test-d/jsonify.ts

```diff
@@ -123,17 +123,85 @@ const nonJsonWithInvalidToJSON = new NonJsonWithInvalidToJSON();
 expectNotAssignable<JsonValue>(nonJsonWithInvalidToJSON);
 expectNotAssignable<JsonValue>(nonJsonWithInvalidToJSON.toJSON());
 
-// Special cases of Array with `undefined` member
-// `[undefined]` is not JSON because it contains non JSON value `undefined`
-// However `JSON.parse(JSON.stringify())` transforms array members of `undefined` to `null`
-expectNotAssignable<JsonValue>([undefined]);
-declare const parsedStringifiedArrayWithUndefined1: Jsonify<undefined[]>; // = JSON.parse(JSON.stringify([undefined]));
-expectType<null[]>(parsedStringifiedArrayWithUndefined1);
-expectAssignable<JsonValue>(parsedStringifiedArrayWithUndefined1);
-expectNotAssignable<JsonValue>([undefined, 1]);
-declare const parsedStringifiedArrayWithUndefined2: Jsonify<[undefined, number]>;
-expectType<Array<number | null>>(parsedStringifiedArrayWithUndefined2);
-expectAssignable<JsonValue>(parsedStringifiedArrayWithUndefined2);
+// Not jsonable types; these types behave differently when used as plain values, as members of arrays and as values of objects
+declare const undef: undefined
```

Comment:
Don't use abbreviations for the variable names. Applies to all of them.

Reply:
Fixed all of them except `fn`, can't use `function` since it's a reserved keyword.